storage/namespace.go

```diff
@@ -234,10 +268,27 @@ func newDatabaseNamespace(
 	}
 
 	n.initShards(nopts.NeedsBootstrap())
+	go n.reportStatusLoop()
```

Comment:
Could you add a leaktest check for ns.Open()+ns.Close() to validate we don't leak this in the future

Reply:
Sure thing, can do.

Reply:
Hey so this is already covered with `TestNamespaceCloseWithShard`.  It checks no more goroutines are running after a namespace close.

Reply:
Separated into two tests:

Reply:
Noice 👍